src/redux/deviseRedux.tsx

```diff
@@ -1,20 +1,24 @@
-/* selectors */
-export const getDevice = ({ device }) => device;
+import { Store } from "./initialState";
+
 
-/* action name creator */
-const reducerName = "device";
-const createActionName = (name) => `app/${reducerName}/${name}`;
+/* selectors */
+export const getDevice = ({ device }:Store) => device;
 
-/* action types */
-const WINDOW_RESIZE = createActionName("WINDOW_RESIZE");
 
 /* action creators */
-export const windowResize = (payload) => ({ type: WINDOW_RESIZE, payload });
+export const windowResize = (payload: boolean) => ({ type: "WINDOW_RESIZE", payload });
+
+//action type
+interface DeviceType {
+  type: "WINDOW_RESIZE"
+  payload: boolean
+}
+
 
 /* reducer */
-export default function reducer(statePart = [], action = {}) {
+export default function reducer(statePart = [], action: DeviceType) {
```

Comment:
Don't forget stating `statePart` type. 

And also the same gripe about `ActionType` as in the other case.

Reply:
Btw, setting the type of state will also reveal the mistake you've made